ksql-rest-app/src/test/java/io/confluent/ksql/api/BasicAuthTest.java

```diff
@@ -0,0 +1,230 @@
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+import io.confluent.ksql.api.server.ApiServerConfig;
+import io.confluent.ksql.test.util.TestBasicJaasConfig;
+import io.confluent.ksql.util.VertxCompletableFuture;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.client.HttpRequest;
+import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.ext.web.client.WebClient;
+import java.util.Map;
+import java.util.function.Consumer;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@RunWith(CoreApiTestRunner.class)
+public class BasicAuthTest extends ApiTest {
+
+  protected static final Logger log = LoggerFactory.getLogger(BasicAuthTest.class);
+
+  private static final String PROPS_JAAS_REALM = "KsqlServer-Props";
+  private static final String KSQL_RESOURCE = "ksql-user";
+  private static final String OTHER_RESOURCE = "not-ksql";
+  private static final String USER_WITH_ACCESS = "harry";
+  private static final String USER_WITH_ACCESS_PWD = "changeme";
+  private static final String USER_WITHOUT_ACCESS = "john";
+  private static final String USER_WITHOUT_ACCESS_PWD = "bad";
+  private static final String USER_WITH_INCORRECT_ROLE = "maud";
+  private static final String USER_WITH_INCORRECT_ROLE_PWD = "1234";
+
+  @ClassRule
+  public static final TestBasicJaasConfig JAAS_CONFIG = TestBasicJaasConfig
+      .builder(PROPS_JAAS_REALM)
+      .addUser(USER_WITH_ACCESS, USER_WITH_ACCESS_PWD, KSQL_RESOURCE)
+      .addUser(USER_WITH_INCORRECT_ROLE, USER_WITH_INCORRECT_ROLE_PWD, OTHER_RESOURCE)
+      .build();
+
+  @Override
+  protected ApiServerConfig createServerConfig() {
+    ApiServerConfig config = super.createServerConfig();
+    Map<String, Object> origs = config.originals();
+    origs.put(
+        ApiServerConfig.AUTHENTICATION_METHOD_CONFIG,
+        ApiServerConfig.AUTHENTICATION_METHOD_BASIC);
+    origs.put(
+        ApiServerConfig.AUTHENTICATION_REALM_CONFIG,
+        PROPS_JAAS_REALM
+    );
+    origs.put(
+        ApiServerConfig.AUTHENTICATION_ROLES_CONFIG,
+        KSQL_RESOURCE
+    );
+    return new ApiServerConfig(origs);
+  }
+
+  @Override
+  protected HttpResponse<Buffer> sendRequest(final WebClient client, final String uri,
+      final Buffer requestBody)
+      throws Exception {
+    return sendRequestWithCreds(client, uri, requestBody, USER_WITH_ACCESS, USER_WITH_ACCESS_PWD);
+  }
+
+  @Override
+  protected void streamRequest(
+      final WebClient client,
+      final String uri,
+      final Consumer<HttpRequest<Buffer>> requestSender) {
+    requestSender.accept(
+        client.post(uri)
+            .basicAuthentication(USER_WITH_ACCESS, USER_WITH_ACCESS_PWD)
+    );
+  }
+
```

Comment:
I think it would also be good to add a test that checks request is rejected if no creds are provided (not bad creds)

Reply:
Done.